### PR TITLE
Adding default analytics publisher

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java [hunk 1]
```diff
@@ -235,6 +235,29 @@ public synchronized void setMaxEvents(int maxEvents) {
...
+    public int getMaxEvents() {
+        return maxEvents;
+    }
+
+    /**
+     * Returns number of stored events.
+     *
+     * @return Number of stored events.
+     */
+    public int getNumStoredEvents() {
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/store/EventStoreManager.java]: Added a couple of convenience APIs to return how many events can be stored and maximum number of events configured.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java [hunk 2]
```diff
@@ -118,6 +124,12 @@ public static synchronized SalesforceAnalyticsManager getInstance(UserAccount ac
             instance = new SalesforceAnalyticsManager(account, communityId);
             INSTANCES.put(uniqueId, instance);
         }
+
+        // Adds a handler for publishing if not already active.
+        if (!PUBLISH_HANDLER_ACTIVE) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java]: This ensures we don't kick off handlers when one exists already.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java [hunk 3]
```diff
@@ -184,6 +196,24 @@ public static synchronized void changePasscode(String oldPass, String newPass) {
     }
 
     /**
+     * Sets the publish frequency, in hours.
+     *
+     * @param publishFrequencyInHours Publish frequency, in hours.
+     */
+    public static synchronized void setPublishFrequencyInHours(int publishFrequencyInHours) {
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/SalesforceAnalyticsManager.java]: API for configurable publish frequency.

**Comment by wmathurin** [comment]
LGTM

**Pull request merged**